src/grouping/group_cache.c

```diff
@@ -7,40 +7,42 @@
 #include "group_cache.h"
 
 CacheGroup *CacheGroupNew() {
-	return NewTrieMap();
+	return raxNew();
 }
 
 void CacheGroupAdd(CacheGroup *groups, char *key, Group *group) {
-	TrieMap_Add(groups, key, strlen(key), group, NULL);
+	raxInsert(groups, (unsigned char *)key, strlen(key), group, NULL);
 }
 
 // Retrives a group,
 // Sets group to NULL if key is missing.
 Group *CacheGroupGet(CacheGroup *groups, char *key) {
-	Group *g = TrieMap_Find(groups, key, strlen(key));
-	if(g == TRIEMAP_NOTFOUND) return NULL;
+	Group *g = raxFind(groups, (unsigned char *)key, strlen(key));
+	if(g == raxNotFound) return NULL;
 	return g;
 }
 
 void FreeGroupCache(CacheGroup *groups) {
-	TrieMap_Free(groups, (void (*)(void *))FreeGroup);
+	raxFreeWithCallback(groups, (void (*)(void *))FreeGroup);
 }
 
-// Returns an iterator to scan entire group cache
-CacheGroupIterator *CacheGroupIter(CacheGroup *groups) {
-	return TrieMap_Iterate(groups, "", 0);
+// Populates an iterator to scan entire group cache
+void CacheGroupIter(CacheGroup *groups, CacheGroupIterator *iter) {
+	raxStart(iter, groups);
```

Comment:
I'd prefer for this function to return an iterator, but doing so causes errors when later calling `raxStop(&iter)`:

```
==6216== Invalid free() / delete / delete[] / realloc()
==6216==    at 0x4C30D3B: free (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==6216==    by 0x10D876: raxStop (rax.c:1803)
==6216==    by 0x108C36: main (empty_test.c:20)
==6216==  Address 0x1ffefff0e0 is on thread 1's stack
==6216==  352 bytes below stack pointer
```

I wrote a toy program to reproduce this, because it was confusing to me.

`raxStart` is supposed to be used to initialize an iterator just once. The `iter_initialized` bool on OpAggregate should guarantee that this function is only called once.

I do think this area could be dramatically cleaned up.